src/uct/sm/mm/mm_posix.c

```diff
@@ -79,14 +90,117 @@ static ucs_status_t uct_posix_test_mem(size_t length, int shm_fd)
     return status;
 }
 
+static size_t uct_posix_get_path_len(uct_pd_h pd)
+{
+    uct_mm_pd_t *mm_pd = ucs_derived_of(pd, uct_mm_pd_t);
+    uct_posix_pd_config_t *posix_config = ucs_derived_of(mm_pd->config,
+                                                         uct_posix_pd_config_t);
+
+    if (posix_config->use_shm_open == UCS_YES) {
+        return 0;
+    } else {
+        return 1 + strlen(posix_config->path);
+    }
+}
+
+static ucs_status_t uct_posix_shm_open(char *file_name, size_t length, int *shm_fd)
+{
+    ucs_status_t status;
+
+    /* Create shared memory object and set its size */
+    *shm_fd = shm_open(file_name, O_CREAT | O_RDWR | O_EXCL,
+                       UCT_MM_POSIX_SHM_OPEN_MODE);
+    if (*shm_fd == -1) {
+        ucs_error("Error returned from shm_open %m. File name is: %s",
+                  file_name);
+        status = UCS_ERR_SHMEM_SEGMENT;
+        goto err;
+    }
+    if (ftruncate(*shm_fd, length) == -1) {
+        ucs_error("Error returned from ftruncate %m");
+        status = UCS_ERR_SHMEM_SEGMENT;
+        goto err_shm_unlink;
+    }
+
+    return UCS_OK;
+
+err_shm_unlink:
+    close(*shm_fd);
+    if (shm_unlink(file_name) != 0) {
+        ucs_warn("unable to unlink the shared memory segment");
+    }
+err:
+    return status;
+}
+
+static ucs_status_t uct_posix_open(char *file_name, size_t length, int *shm_fd)
+{
+    ucs_status_t status;
+
+    /* use open with the given path */
+    *shm_fd = open(file_name, O_CREAT | O_RDWR | O_EXCL, UCT_MM_POSIX_SHM_OPEN_MODE);
+    if (*shm_fd == -1) {
+        ucs_error("Error returned from open %m . File name is: %s", file_name);
+        status = UCS_ERR_SHMEM_SEGMENT;
+        goto err;
+    }
+
+    if (ftruncate(*shm_fd, length) == -1) {
+        ucs_error("Error returned from ftruncate %m");
+        status = UCS_ERR_SHMEM_SEGMENT;
+        goto err_close;
+    }
+
+    return UCS_OK;
+
+err_close:
+    close(*shm_fd);
+err:
+    return status;
+}
+
+static ucs_status_t
+uct_posix_open_backing_file(char **file_name, uint64_t *uuid, uct_posix_pd_config_t *config,
+                            size_t length, int *shm_fd, const char **path_p)
+{
+    ucs_status_t status;
+
+    if (config->use_shm_open != UCS_NO) {
+        sprintf(*file_name, "/ucx_shared_mr_uuid_%zu", *uuid >> 2);
```

Comment:
if the user provides a very long path, we should fail to open the file (becase presumably this path is not valid, since all valid paths are smaller than NAME_MAX), but we should not get a segfault.
Besides, this is bad practice to use string functions which do not have an upper limit, unless you are absolutely sure (programmatically) that you know the string length.